### PR TITLE
Update goals for Java milestones

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: handbook/engineering/code-intelligence/goals.md [hunk 1]
```diff
@@ -70,18 +70,18 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
 **Milestones:**
 
 1. 🔄 Sourcegraph Cloud has up-to-date precise code intelligence for 5 Gradle repositories.
-    - 🔄 Manual configuration in a forked repository.
-    - 🔄 Gradle plugin implementation.
+    - ✅ Automatic inference for projects that use the Java plugin
+    - Automatic inference for projects that use the Android plugin
     - Publish Gradle plugin.
```
Comment (olafurpg) [inline handbook/engineering/code-intelligence/goals.md:75]: The gradle/maven plugin may not be necessary, maybe we should remove them from the roadmap. I don't think we will implement plugins unless we see that they're necessary for customer adoption

**Review by macraig — APPROVED** [review]
Thanks for updating!

**Review by Strum355 — APPROVED** [review]
🚀

**Review by scalabilitysolved — APPROVED** [review]
Looks good to me :rocket:

**Pull request merged**